pallets/foreign-investments/src/swaps.rs

```diff
@@ -152,13 +129,13 @@ impl<T: Config> Swaps<T> {
 					who.clone(),
 					new_swap.currency_in,
 					new_swap.currency_out,
-					new_swap.amount_in,
-					T::BalanceRatio::one(),
+					new_swap.amount_out,
+					OrderRatio::Market,
```

Comment:
**CLARIFICATION**

Isn't it problematic that the `SwapOf<T>` has a directionality when reaching this method - `currency_out` &rarr; `currency_in`, but the returned `SwapStatus` might have a changed directionality. 

So depending on what happened internally here, the `SwapStatus` fields `swapped` and `pending` refer to different currencies...

Reply:
From the point of view of the caller, that is "updating" the swap here:

`swapped` and `pending` refer to 
* if no order &rarr; `currency_in`
* if order same direction &rarr; `currency_in`
* if order reverse direction
   * if `reverse_amount_in` greater &rarr; `currency_out`
   * if `reverse_amount_in` equal &rarr; `currency_in`
   * if `reverse_amount_in` smaller &rarr; `currency_in`

Reply:
@lemunozm assume the directionality has changed. Wouldn't [this line (same as code below)](https://github.com/centrifuge/centrifuge-chain/blob/12d66f6e1a2ec557ff943af68ded7b3d020746ae/pallets/foreign-investments/src/impls.rs#L41)

```rust
let swapped_foreign_amount = foreign_amount.ensure_sub(status.pending)?;
```

be problematic?

Reply:
> - if reverse_amount_in greater → currency_out

If I'm not wrong, this is `currnecy_in`, so all cases return `currency_in` as `swapped` and `currency_out` as `pending`.

So, in the referenced line, `status.pending` from an increase swap is `foreign_currency`, and it's subtracted with `foreing_amount`, so both currencies match

Reply:
Yeah, that resulted in my misunderstanding of the return value. It is always the same directionality then! Thanks for claryfiuyng.